applications/tari_console_wallet/src/ui/state/app_state.rs

```diff
@@ -181,6 +184,30 @@ impl AppState {
         }
     }
 
+    pub async fn check_connectivity(&mut self) {
+        let update = match self.connectivity_check_cooldown {
+            Some(last_update) => last_update.elapsed() > self.config.connectivity_check_interval,
+            None => {
+                self.connectivity_check_cooldown = Some(Instant::now());
+                false
+            },
+        };
+
+        if update && self.wallet_connectivity.get_connectivity_status() == OnlineStatus::Offline {
+            let current = self.get_selected_base_node();
+            let list = self.get_base_node_list().clone();
+            let mut index: usize = list.iter().position(|(_, p)| p == current).unwrap_or_default();
+            if index == list.len() {
```

Comment:
`index` can never equal the length of the list because it is an index in the list.
Should be 
```suggestion
            if index == list.len() - 1 {
```

or

```rust
let index = (index + 1) % list.len();
```

In both of these you need to make sure before that list isn't empty (i.e. `- 1 ` underflows or divide by zero) - pretty sure that the wallet exits with an error if that is the case, but thought I'd mention it in case